pkg/limits/ingest_limits.go

```diff
@@ -363,10 +380,9 @@ func (s *IngestLimits) ExceedsLimits(_ context.Context, req *proto.ExceedsLimits
 		bucketCutoff = lastSeenAt.Add(-s.cfg.RateWindow).UnixNano()
 		// Calculate the max active streams per tenant per partition
 		maxActiveStreams = uint64(s.limits.MaxGlobalStreamsPerUser(req.Tenant) / s.cfg.NumPartitions)
-		// Create a map of streams per partition
-		streams = make(map[int32][]Stream)
 	)
 
+	streams := make([]*proto.StreamMetadata, 0, len(req.Streams))
```

Comment:
This looks like a mistake to me? Why copy the streams from the request into a new slice?